File: reference-states/contracts-java/src/main/java/com.example/contract/SanctionedEntitiesContract.java
package com.example.contract;

import com.example.state.SanctionedEntities;
import net.corda.core.contracts.Command;
import net.corda.core.contracts.CommandData;
import net.corda.core.contracts.Contract;
import net.corda.core.transactions.LedgerTransaction;

import static net.corda.core.contracts.ContractsDSL.requireThat;

public class SanctionedEntitiesContract implements Contract {
    public static String SANCTIONS_CONTRACT_ID = "com.example.contract.SanctionedEntitiesContract";


    /**
     * The verify() function of all the states' contracts must not throw an exception for a transaction to be
     * considered valid.
     */
    @Override
    public void verify(LedgerTransaction tx) throws IllegalArgumentException {
        Command command = tx.commandsOfType(Commands.class).get(0);

        if(command.getValue() instanceof  Commands.Create){
            requireThat(require -> {
                require.using("when creating a sanctions list there should be no inputs", tx.getInputStates().isEmpty());
                require.using("when creating a sanctions list there should be one output", tx.outputsOfType(SanctionedEntities.class).size() == 1);
                SanctionedEntities out = tx.outputsOfType(SanctionedEntities.class).get(0);
                require.using("The issuer of the sanctions list must sign", command.getSigners().contains(out.getIssuer().getOwningKey()));
                return null;
            }) ;
        }else if(command.getValue() instanceof Commands.Update){
            requireThat(require -> {
                require.using("There must be exactly one input Sanctions List when updating", tx.inputsOfType(SanctionedEntities.class).size() == 1);
                require.using("There must be exactly one output Sanctions List when updating", tx.outputsOfType(SanctionedEntities.class).size() == 1);
                SanctionedEntities input = tx.inputsOfType(SanctionedEntities.class).get(0);
                SanctionedEntities output = tx.outputsOfType(SanctionedEntities.class).get(0);
                require.using("The issuer must remain the same across an update", input.getIssuer().equals(output.getIssuer()));
                return null;
            });
        }
    }


    public interface Commands extends CommandData {
        class Create implements Commands {};
        class Update implements  Commands {};
    }
}

File: reference-states/reference-example-contracts/src/main/kotlin/com/example/contract/SanctionedEntitiesContract.kt
package com.example.contract

import com.example.state.SanctionedEntities
import net.corda.core.contracts.CommandData
import net.corda.core.contracts.Contract
import net.corda.core.contracts.requireThat
import net.corda.core.transactions.LedgerTransaction

class SanctionedEntitiesContract : Contract {
    companion object {
        @JvmStatic
        val SANCTIONS_CONTRACT_ID = "com.example.contract.SanctionedEntitiesContract"
    }

    /**
     * The verify() function of all the states' contracts must not throw an exception for a transaction to be
     * considered valid.
     */
    override fun verify(tx: LedgerTransaction) {
        val command = tx.commandsOfType(Commands::class.java).single()

        if (command.value is Commands.Create) {
            requireThat {
                "when creating a sanctions list there should be no inputs" using (tx.inputStates.isEmpty())
                "when creating a sanctions list there should be one output" using (tx.outputsOfType(SanctionedEntities::class.java).size == 1)
                val out = tx.outputsOfType(SanctionedEntities::class.java).single()
                "The issuer of the sanctions list must sign" using (out.issuer.owningKey in command.signers)
            }

        } else if (command.value is Commands.Update) {
            requireThat {
                "There must be exactly one input Sanctions List when updating" using (tx.inputsOfType(SanctionedEntities::class.java).size == 1)
                "There must be exactly one output Sanctions List when updating" using (tx.outputsOfType(
                    SanctionedEntities::class.java
                ).size == 1)
                val input = tx.inputsOfType<SanctionedEntities>().single()
                val output = tx.outputsOfType<SanctionedEntities>().single()
                "The issuer must remain the same across an update" using (input.issuer == output.issuer)
            }
        }
    }

    /**
     * This contract only implements one command, Create.
     */
    interface Commands : CommandData {
        object Create : Commands
        object Update : Commands
    }
}
